nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/clients/OAuth2ClientServiceImpl.java

```diff
@@ -66,6 +72,44 @@ public OAuth2Client getClient(String clientId) {
         return queryClients().stream().map(OAuth2Client::fromDocumentModel).collect(Collectors.toList());
     }
 
+    @Override
+    public OAuth2Client create(OAuth2Client oAuth2Client) {
+        Objects.requireNonNull(oAuth2Client, "oAuth2Client is required");
```

Comment:
Not needed if you call `OAuth2Client#validate` directly here as suggested above.